src/main/java/net/minestom/server/component/DataComponentMapImpl.java

```diff
@@ -68,6 +70,26 @@ public boolean has(@NotNull DataComponentMap prototype, @NotNull DataComponent<?
         return new DataComponentMapImpl(newComponents);
     }
 
+    @Override
+    public void forEach(@NotNull BiConsumer<DataComponent<?>, Object> consumer) {
```

Comment:
These are not good and need to be resolved before this is merged.

Some options (none of which I am not in love with currently):
* Change the consumer to return ints and let the caller handle it
* track the ID->DataComponent mapping internally

I think tracking the id mapping is the most obvious option, but I am not that excited about increasing the memory footprint of ItemStack :(